lib/zuora/api.rb

```diff
@@ -11,24 +11,39 @@ module Zuora
   # @return [Config]
   def self.configure(opts={})
     Api.instance.config = Config.new(opts)
-    HTTPI.logger = opts[:logger]
-    HTTPI.log = opts[:logger] ? true : false
-    Savon.configure do |savon|
-      savon.logger = opts[:logger]
-      savon.log = opts[:logger] ? true : false
+    if Api.instance.config.sandbox
+      Api.instance.sandbox!
+    elsif Api.instance.config.services
+      Api.instance.set_endpoint Api.instance.config.custom_url
     end
   end
 
   class Api
+    I18n.enforce_available_locales = false
+
     # @return [Savon::Client]
-    attr_accessor :client
+    def client
+      @client ||= make_client
+    end
 
     # @return [Zuora::Session]
     attr_accessor :session
 
     # @return [Zuora::Config]
     attr_accessor :config
 
+    # Zuora::API Config options
+    # @return [Hash]
+    attr_accessor :options
+
+    WSDL = File.expand_path('../../../wsdl/zuora.a.78.0.wsdl', __FILE__)
```

Comment:
In the old file, we had this check, may be will have to retain if condition
```
wsdl_path = if config && config.wsdl_path
              config.wsdl_path
            else
              File.expand_path('../../../wsdl/zuora.a.38.0.wsdl', __FILE__)
            end
```

Reply:
Commented out the above suggestion in the commit [46b7ecf](https://github.com/Sage/zuora/pull/18/commits/46b7ecfb2dce048c68e2445a637874284b558a5e), pending when we understand if we would need to retain the above pattern